EventListener/JmsPreSerializeListener.php

```diff
@@ -44,13 +52,23 @@ public function onPreSerialize(ObjectEvent $event)
             foreach ($reflectionClass->getProperties() as $property) {
                 $liipAnnotation = $this->annotationReader->getPropertyAnnotation($property, LiipImagineSerializableField::class);
                 $property->setAccessible(true);
-
                 if ($liipAnnotation instanceof LiipImagineSerializableField && ($value = $property->getValue($object)) && !is_array($value)) {
                     $vichField = $liipAnnotation->getVichUploaderField();
 
+                    $uriComponents = explode($value, '/');
+                    $cacheKey      = $vichField.array_pop($uriComponents);
+
+                    if (in_array($cacheKey, $this->cache)) {
+                        continue;
+                    }
+
                     if (!$liipAnnotation->getVirtualField()) {
+                        $this->cache[] = $cacheKey;
+
                         $property->setValue($object, $this->serializeValue($liipAnnotation, $object, $value));
                     } elseif ($vichField && array_key_exists('vichUploaderSerialize', $this->config) && $this->config['vichUploaderSerialize']) {
+                        $this->cache[] = $cacheKey;
```

Comment:
You can remove $this->cache[] = $cacheKey; in this and previous if and put it once in the end. At 78 line.

Reply:
I did it twice because it's not just a simple `if/else`, but `if/elseif`, we don't want to add the key to cache if none of the condition is met.

Reply:
Ok.